abci/types/pubkey.go

```diff
@@ -22,3 +24,17 @@ func Ed25519ValidatorUpdate(pk []byte, power int64) ValidatorUpdate {
 		Power:  power,
 	}
 }
+
+func BLS12381ValidatorUpdate(pk []byte, power int64) ValidatorUpdate {
+	pke := bls12381.PubKey(pk)
+	pkp, err := cryptoenc.PubKeyToProto(pke)
+	if err != nil {
+		panic(err)
```

Comment:
Is it ok to panic here? What could be a source of errors here?

Reply:
I copied the exact same logic as in Edwards keys.

Reply:
Discussed on call. It's good. The error could be possible only in case if we introduce a new key type and don't handle it in `PubKeyToProto` method.